src/query/plan/operator.hpp

```diff
@@ -2199,6 +2199,8 @@ class CallProcedure : public memgraph::query::plan::LogicalOperator {
   Expression *memory_limit_{nullptr};
   size_t memory_scale_{1024U};
   bool is_write_;
+  mutable utils::MonotonicBufferResource monotonic_memory{1024UL * 1024UL};
```

Comment:
Do we need to have this inside the class CallProcedures?

Reply:
Yes, since it is specific for CallProcedure and it needs to live as long as CallProcedure query execution lives. I don't have any better idea due to this requirements.